parquet-hadoop/src/main/java/org/apache/parquet/hadoop/DictionaryStream.java

```diff
@@ -0,0 +1,61 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+import org.apache.parquet.bytes.BytesInput;
+import org.apache.parquet.column.Encoding;
+import org.apache.parquet.column.page.DictionaryPage;
+import org.apache.parquet.format.DictionaryPageHeader;
+import org.apache.parquet.format.PageHeader;
+import org.apache.parquet.format.PageType;
+import org.apache.parquet.format.Util;
+
+/**
+ * Data stream to only read dictionary page
+ */
+public class DictionaryStream extends ByteArrayInputStream {
+
+  public DictionaryStream(byte[] data, int offset) {
+    super(data);
+    this.pos = offset;
+  }
+
+  public DictionaryPage readDictionaryPage() throws IOException {
+    PageHeader pageHeader = Util.readPageHeader(this);
+    int uncompressedPageSize = pageHeader.getUncompressed_page_size();
+    int compressedPageSize = pageHeader.getCompressed_page_size();
+    if (pageHeader.type == PageType.DICTIONARY_PAGE) {
+      DictionaryPageHeader dicHeader = pageHeader.getDictionary_page_header();
+      return new DictionaryPage(getBytesInput(compressedPageSize),
+                                uncompressedPageSize,
+                                dicHeader.getNum_values(),
+                                Encoding.valueOf(dicHeader.getEncoding().name()));
+    }
+    return null;
+  }
```

Comment:
let's not duplicate that code.
You can refactor some of the code here:
https://github.com/apache/parquet-mr/blob/c3819688c48480ec75a9563c71f18ea755e34620/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnChunkPageReadStore.java#L129
So that this logic is reused.
Also make the naming consistent